audio_player.cpp

```diff
@@ -382,7 +380,7 @@ static esp_err_t aplay_file(audio_instance_t *i, FILE *fp)
                 bytes_to_write,
                 i->output.frame_count);
 
-            i2s_write(i->port, i->output.samples, bytes_to_write, &i2s_bytes_written, portMAX_DELAY);
+            i->config.write_fn((char *)i->output.samples, bytes_to_write, &i2s_bytes_written, portMAX_DELAY);
```

Comment:
We can drop the cast here?

Reply:
> We can drop the cast here?
Add a condition to judge whether this function is empty. or other ways?

Reply:
> No I mean isn't it fine without the (char*)?

ok, got It